uaclient/cli.py

```diff
@@ -1282,44 +1238,13 @@ def action_auto_attach(args, *, cfg):
     if cfg.is_attached:
         raise exceptions.AlreadyAttachedOnPROError()
 
-    disable_auto_attach = util.is_config_value_true(
-        config=cfg.cfg, path_to_value="features.disable_auto_attach"
-    )
-    if disable_auto_attach:
-        msg = "Skipping auto-attach. Config disable_auto_attach is set."
-        logging.debug(msg)
-        print(msg)
+    skip_auto_attach = actions.should_disable_auto_attach(cfg)
+    if skip_auto_attach:
         return 0
 
-    instance = None  # type: Optional[AutoAttachCloudInstance]
-    try:
-        instance = identity.cloud_instance_factory()
-    except exceptions.CloudFactoryError as e:
-        if cfg.is_attached:
-            # We are attached on non-Pro Image, just report already attached
-            raise exceptions.AlreadyAttachedError(cfg)
-        if isinstance(e, exceptions.CloudFactoryNoCloudError):
-            raise exceptions.UserFacingError(
-                messages.UNABLE_TO_DETERMINE_CLOUD_TYPE
-            )
-        if isinstance(e, exceptions.CloudFactoryNonViableCloudError):
-            raise exceptions.UserFacingError(messages.UNSUPPORTED_AUTO_ATTACH)
-        if isinstance(e, exceptions.CloudFactoryUnsupportedCloudError):
-            raise exceptions.NonAutoAttachImageError(
-                messages.UNSUPPORTED_AUTO_ATTACH_CLOUD_TYPE.format(
-                    cloud_type=e.cloud_type
-                )
-            )
-        # we shouldn't get here, but this is a reasonable default just in case
-        raise exceptions.UserFacingError(
-            messages.UNABLE_TO_DETERMINE_CLOUD_TYPE
-        )
-
+    instance = actions.get_cloud_instance(cfg)
     if not instance:
-        # we shouldn't get here, but this is a reasonable default just in case
-        raise exceptions.UserFacingError(
-            messages.UNABLE_TO_DETERMINE_CLOUD_TYPE
-        )
+        return 0
```

Comment:
I don't think we need this

Reply:
To clarify, we don't need the `if not instance:` check because it can't be None

Reply:
True, we will just raise an exception.